lib/autoproj/package_managers/bundler_manager.rb

```diff
@@ -88,11 +88,82 @@ def initialize_environment
                     end
                 end
 
-                if bundle_rubylib = discover_bundle_rubylib(silent_errors: true)
+                apply_build_config(prefix_gems)
+
+                if (bundle_rubylib = discover_bundle_rubylib(silent_errors: true))
                     update_env_rubylib(bundle_rubylib, system_rubylib)
                 end
             end
 
+            # Enumerate the per-gem build configurations
+            def per_gem_build_config
+                ws.config.get('bundler.build', {})
+            end
+
+            # Set the build configuration for the given gem
+            #
+            # This is meant to be used from the Autoproj configuration files,
+            # e.g. overrides.rb or package configuration
+            def self.configure_build_for(ws, gem_name, build_config)
```

Comment:
Could we move `ws` to last and have a default to `Autoproj.workspace`?

Reply:
Makes a lot of sense. :+1:

Reply:
Done